bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+                }
+                break;
+            case CONFIG_LOCATION_TYPE_IATA_CODE:
+                if (StringUtils.trimToNull(config.iataCode) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IATA code is not set");
+                    validConfig = false;
+                } else {
+                    sb.append("&iataCode=").append(config.iataCode.replace(" ", "").toUpperCase());
+                }
+                break;
+        }
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        return sb.toString();
+    }
+
+    /*
+     * Set either Imperial or Metric SI for the API call
+     */
+    private String getUnitsQueryString() {
+        return isImperial() ? "e" : "s";
+    }
+
+    /*
+     * Build the URL for requesting the PWS current observations
+     */
+    private @Nullable String buildPwsUrl() {
+        if (StringUtils.isEmpty(config.pwsStationId)) {
+            return null;
+        }
+        StringBuilder sb = new StringBuilder(BASE_PWS_URL);
+        // Set to use Imperial units. UoM will convert to the other units
+        sb.append("?units=e");
+        // Set response type as JSON
+        sb.append("&format=json");
+        // Set PWS station Id from config
+        sb.append("&stationId=").append(config.pwsStationId);
+        // Set API key from config
+        sb.append("&apiKey=").append(config.apiKey);
+        String url = sb.toString();
+        logger.debug("PWS observations URL is {}", url.replace(config.apiKey, REPLACE_API_KEY));
+        return url;
+    }
+
+    /*
+     * Determine the units configured in the system
+     */
+    private boolean isImperial() {
+        return systemOfUnits instanceof ImperialUnits ? true : false;
+    }
+
+    private void refreshForecast() {
+        if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+            logger.debug("Handler: Can't refresh forecast because thing configuration is incomplete");
+            return;
+        }
+        logger.debug("Handler: Requesting forecast from The Weather Company API");
+        String response = executeApiRequest(forecastUrl);
+        if (response == null) {
+            return;
+        }
+        try {
+            logger.trace("Handler: Parsing forecast response: {}", response);
+            Forecast forecast = gson.fromJson(response, Forecast.class);
+            logger.debug("Handler: Successfully parsed daily forecast response object");
+            if (thing.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+            updateDailyForecast(forecast);
+            updateDaypartForecast(forecast.daypart);
+        } catch (JsonSyntaxException e) {
+            logger.debug("Handler: Error parsing daily forecast response object: {}", e.getMessage(), e);
```

Comment:
AFAIK it will show the message of you pass the complete exception, so IMHO not needed to pass the message separately

Reply:
Message removed.